### ISSUE TITLE
DBNet, ICDAR2015, dist_test, get h-mean 0.00

### ISSUE THREAD

**Issue by Bovey0809** [open-mmlab/mmocr#17]
The demo part is fine with the pretrained weight, but wehn I try for evaluation, the results turn to be wrong.

```bash
./tools/dist_test.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py checkpoints/dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth 8 --eval hmean-iou
```

![image](https://user-images.githubusercontent.com/10627971/113993178-bc78b900-9886-11eb-8777-56a0653635bd.png)

**Comment by innerlee** [comment]
Thanks for the report! We will check

**Comment by quincylin1** [comment]
> The demo part is fine with the pretrained weight, but wehn I try for evaluation, the results turn to be wrong.
> 
> ```shell
> ./tools/dist_test.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py checkpoints/dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth 8 --eval hmean-iou
> ```
> 
> ![image](https://user-images.githubusercontent.com/10627971/113993178-bc78b900-9886-11eb-8777-56a0653635bd.png)

We have tested locally with the following command. The checkpoint was downloaded from openmmlab: ./tools/dist_test.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py  https://download.openmmlab.com/mmocr/textdet/dbnet/dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.pth 1 --eval hmean-iou
and managed to obtain the following results:
![image](https://user-images.githubusercontent.com/33197366/114007132-a3c2d000-9893-11eb-89ca-332b003ffa42.png)
It would be helpful if you could provide the steps that you implemented so we could reproduce the error you encountered and try to debug it. Thanks!

**Comment by Bovey0809** [comment]
> > The demo part is fine with the pretrained weight, but wehn I try for evaluation, the results turn to be wrong.
> > ```shell
> > ./tools/dist_test.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py checkpoints/dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth 8 --eval hmean-iou
> > ```
> > 
> > 
> > ![image](https://user-images.githubusercontent.com/10627971/113993178-bc78b900-9886-11eb-8777-56a0653635bd.png)
> 
> We have tested locally with the following command. The checkpoint was downloaded from openmmlab:
> 
> ```shell
> ./tools/dist_test.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py  https://download.openmmlab.com/mmocr/textdet/dbnet/dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.pth 1 --eval hmean-iou
> ```
> 
> and managed to obtain the following results:
> ![image](https://user-images.githubusercontent.com/33197366/114007132-a3c2d000-9893-11eb-89ca-332b003ffa42.png)
> It would be helpful if you could provide the steps that you implemented so we could reproduce the error you encountered and try to debug it. Thanks!

Thanks for your quickly reply.
Here is what I got from the same command.
![image](https://user-images.githubusercontent.com/10627971/114019653-3fa70880-98a1-11eb-8310-36431f9ef5d6.png)

Here are the steps for the failed F1 score.
1. Install the env as the document.
2. Download the ICDAR2015 and arranged the dataset as what I have been told as the doc.
3. Download the pretrained weight from https://mmocr.readthedocs.io/en/latest/textdet_models.html
4. Try for the command "./tools/dist_test.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py checkpoints/dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth 8 --eval hmean-iou"

**Comment by innerlee** [comment]
@Bovey0809 Could you please try testing with 1 gpu? This will rule out problems in distributed codes.

If problems still exists, then the next suspect is data

**Comment by Bovey0809** [comment]
I have tried the one GPU version. After debugging for a while, it seems that the weight dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth" downloaded from "https://mmocr.readthedocs.io/en/latest/textdet_models.html" cannot be loadded into the model, maybe some name conflicts, I will check later.

**Comment by innerlee** [comment]
You can use ckpt url directly in the testing command, as @quincylin1 shows. Pls check if the ckpt is corrupted

**Comment by jeffreykuang** [comment]
> I have tried the one GPU version. After debugging for a while, it seems that the weight dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth" downloaded from "https://mmocr.readthedocs.io/en/latest/textdet_models.html" cannot be loadded into the model, maybe some name conflicts, I will check later.

donot use dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth, which is the pretrained model on synthtext.

Use this model https://download.openmmlab.com/mmocr/textdet/dbnet/dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.pth instead if you want to evaluate dbnet on icdar2015

**Comment by Bovey0809** [comment]
> > I have tried the one GPU version. After debugging for a while, it seems that the weight dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth" downloaded from "https://mmocr.readthedocs.io/en/latest/textdet_models.html" cannot be loadded into the model, maybe some name conflicts, I will check later.
> 
> donot use dbnet_r50dcnv2_fpnc_sbn_2e_synthtext_20210325-aa96e477.pth, which is the pretrained model on synthtext.
> 
> Use this model https://download.openmmlab.com/mmocr/textdet/dbnet/dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.pth instead if you want to evaluate dbnet on icdar2015

Ok, there is still a small gap between our results, yours are 0.829 and mine is 0.796, with the same model(downloaded from the url). Do you have any ideas about it? I'm using the original config from location configs/textdet/dbnet.

**Comment by innerlee** [comment]
Please post the full log (including all outputs from the bash command to the end

**Comment by Bovey0809** [comment]
![image](https://user-images.githubusercontent.com/10627971/114122343-a9b3c200-9922-11eb-9d7f-c3cb65127b8a.png)
```
~/code/mmocr(main*) » ./tools/dist_test.sh configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py  https://download.openmmlab.com/mmocr/textdet/dbnet/dbnet_r50dcnv2_fpnc_sbn_1200e_icdar2015_20210325-91cef9af.pth 1 --eval hmean-iou
loading annotations into memory...
Done (t=0.09s)
creating index...
index created!
[>                                                 ] 12/500, 0.9 task/s, elapsed: 13s, ETA:   540s/home/houbowei/code/mmocr/mmocr/models/textdet/postprocess/wrapper.py:179: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  expanded = np.array(offset.Execute(distance))
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 500/500, 13.4 task/s, elapsed: 37s, ETA:     0s
Evaluating hmean-iou...
thr 0.3, recall：0.773, precision: 0.762, hmean:0.768
thr 0.4, recall：0.773, precision: 0.772, hmean:0.773
thr 0.5, recall：0.773, precision: 0.788, hmean:0.780
thr 0.6, recall：0.771, precision: 0.819, hmean:0.794
thr 0.7, recall：0.755, precision: 0.843, hmean:0.797
```
Here is the command and log.

**Comment by quincylin1** [comment]
![image](https://user-images.githubusercontent.com/33197366/114123939-ecc36480-9925-11eb-8197-271b7d0e7ff3.png)
We ran the same command as yours and we managed to get the results successfully.
Are you sure you have fetched the latest version from origin/main? Upon evaluation the final results should be displayed as a dict on the last line (as shown in the screenshot above).
Please let us know if it still doesn't work with the latest fetch and we are happy to help!

**Comment by Bovey0809** [comment]
After update to the latest code. And install some missing packages.
I successfully get the same result as yours.

PS: My config file is a little bit different from the default one, I have corrected it.
https://github.com/open-mmlab/mmocr/blob/1d1cd9fd495d727b39148bc649c3af2a1ff367bc/configs/textdet/dbnet/dbnet_r50dcnv2_fpnc_1200e_icdar2015.py#L76
I changed it to (1280, 720). That's the reason why my F1 score is 0.2 lower than yours.
Thanks.
